File: studentgrades.js
const prompt = require("prompt-sync")();

let numStudents = parseInt(prompt("Enter number of students: "));
let numSubjects = parseInt(prompt("Enter number of subjects: "));

let subjects = [];
for (let j = 0; j < numSubjects; j++) {
  subjects.push(prompt(`Enter name for subject ${j + 1}: `));
}

let students = [];

for (let i = 0; i < numStudents; i++) {
  let name = prompt(`Enter student ${i + 1} name: `);
  let scores = {};
  for (let j = 0; j < numSubjects; j++) {
    let score;
    do {
      score = parseInt(prompt(`Enter score for ${name} in ${subjects[j]} (0-100): `));
    } while (score < 0 || score > 100);
    scores[subjects[j]] = score;
  }
  students.push({ name, scores });
}

let passCount = {}, failCount = {}, totalScores = {};
subjects.forEach(subject => { passCount[subject] = 0; failCount[subject] = 0; });

students.forEach(student => {
  let total = 0;
  for (let subject of subjects) {
    let score = student.scores[subject];
    total += score;
    if (score >= 50) passCount[subject]++;
    else failCount[subject]++;
  }
  totalScores[student.name] = total;
});

let bestSubject = subjects.reduce((a, b) => passCount[a] > passCount[b] ? a : b);
let worstSubject = subjects.reduce((a, b) => failCount[a] > failCount[b] ? a : b);

let bestStudent = Object.keys(totalScores).reduce((a, b) => totalScores[a] > totalScores[b] ? a : b);
let worstStudent = Object.keys(totalScores).reduce((a, b) => totalScores[a] < totalScores[b] ? a : b);

console.log("\n===== CLASS SUMMARY =====");
console.log(Best Subject: ${bestSubject});
console.log(Worst Subject: ${worstSubject});
console.log(Best Graduating Student: ${bestStudent} (Total: ${totalScores[bestStudent]}));
console.log(Worst Graduating Student: ${worstStudent} (Total: ${totalScores[worstStudent]}));

console.log("\nAll Students:");
students.forEach(s => console.log(${s.name} - Total Score: ${totalScores[s.name]}));

File: StudentGrades.java
import java.util.*;

public class StudentGrades {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Enter number of students: ");
        int numStudents = input.nextInt();

        System.out.print("Enter number of subjects: ");
        int numSubjects = input.nextInt();

        String[] subjects = new String[numSubjects];
        for (int i = 0; i < numSubjects; i++) {
            System.out.print("Enter subject " + (i + 1) + " name: ");
            subjects[i] = input.next();
        }

        String[] studentNames = new String[numStudents];
        int[][] scores = new int[numStudents][numSubjects];

        for (int i = 0; i < numStudents; i++) {
            System.out.print("Enter name of student " + (i + 1) + ": ");
            studentNames[i] = input.next();
            for (int j = 0; j < numSubjects; j++) {
                int score;
                do {
                    System.out.print("Enter score for " + subjects[j] + ": ");
                    score = input.nextInt();
                } while (score < 0 || score > 100);
                scores[i][j] = score;
            }
        }

        int[] totalScores = new int[numStudents];
        int[] passCounts = new int[numSubjects];
        int[] failCounts = new int[numSubjects];

        for (int i = 0; i < numStudents; i++) {
            for (int j = 0; j < numSubjects; j++) {
                int score = scores[i][j];
                totalScores[i] += score;
                if (score >= 50) {
                    passCounts[j]++;
                } else {
                    failCounts[j]++;
                }
            }
        }

        int bestSubjectIndex = 0, worstSubjectIndex = 0;
        for (int j = 1; j < numSubjects; j++) {
            if (passCounts[j] > passCounts[bestSubjectIndex]) bestSubjectIndex = j;
            if (failCounts[j] > failCounts[worstSubjectIndex]) worstSubjectIndex = j;
        }

        int bestStudentIndex = 0, worstStudentIndex = 0;
        for (int i = 1; i < numStudents; i++) {
            if (totalScores[i] > totalScores[bestStudentIndex]) bestStudentIndex = i;
            if (totalScores[i] < totalScores[worstStudentIndex]) worstStudentIndex = i;
        }

        System.out.println("\n=== CLASS SUMMARY ===");
        System.out.println("Best Subject (most passed): " + subjects[bestSubjectIndex]);
        System.out.println("Worst Subject (most failed): " + subjects[worstSubjectIndex]);
        System.out.println("Best Student: " + studentNames[bestStudentIndex] + " with total: " + totalScores[bestStudentIndex]);
        System.out.println("Worst Student: " + studentNames[worstStudentIndex] + " with total: " + totalScores[worstStudentIndex]);

        for (int i = 0; i < numStudents; i++) {
            System.out.print(studentNames[i] + ": ");
            for (int j = 0; j < numSubjects; j++) {
                System.out.print(subjects[j] + "=" + scores[i][j] + " ");
            }
            System.out.println("Total=" + totalScores[i]);
        }
    }
}